aspnetcore/web-api/define-controller.md

```diff
@@ -0,0 +1,113 @@
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api.Pre21/Controllers/PetsController.cs?name=snippet_PetsController&highlight=3)]
+
+A benefit of deriving from `ControllerBase` instead of `Controller` is that IntelliSense displays only Web API-specific members.
+
+::: moniker range=">= aspnetcore-2.1"
+## Annotate class with ApiControllerAttribute
+
+ASP.NET Core 2.1 introduces the `[ApiController]` attribute to denote a Web API controller class. For example:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api/Controllers/ProductsController.cs?name=snippet_ControllerSignature&highlight=2)]
+
+This attribute is commonly coupled with either `ControllerBase` or `Controller` to gain access to useful methods and properties. `ControllerBase` provides access to methods such as [CreatedAtAction](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.createdataction) and [File](/dotnet/api/microsoft.aspnetcore.mvc.controllerbase.file). `Controller` provides access to methods such as [Json](/dotnet/api/microsoft.aspnetcore.mvc.controller.json) and [View](/dotnet/api/microsoft.aspnetcore.mvc.controller.view).
+
+The following sections describe convenience features added by the attribute.
+
+### Automatic HTTP 400 responses
+
+Validation errors automatically trigger an HTTP 400 response. The following code becomes unnecessary in your actions:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api.Pre21/Controllers/PetsController.cs?range=46-49)]
+
+This default behavior is disabled with the following code in *Startup.ConfigureServices*:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api/Startup.cs?name=snippet_ConfigureApiBehaviorOptions&highlight=5)]
+
+### Binding source parameter inference
+
+Inference rules are applied for the default data sources of action parameters. The binding source attributes behave as follows:
+
+* **[[FromBody]](/dotnet/api/microsoft.aspnetcore.mvc.frombodyattribute)** is inferred for complex type parameters. An exception to this rule is any complex, built-in type with a special meaning, such as [IFormCollection](/dotnet/api/microsoft.aspnetcore.http.iformcollection) and [CancellationToken](/dotnet/api/system.threading.cancellationtoken). The binding source inference code ignores those special types. When an action has more than one parameter explicitly specified (via `[FromBody]`) or inferred as bound from the request body, an exception is thrown. For example, the following action signatures cause an exception:
+
+[!code-csharp[](../web-api/define-controller/samples/WebApiSample.Api/Controllers/TestController.cs?name=snippet_ActionsCausingExceptions)]
+
+* **[[FromForm]](/dotnet/api/microsoft.aspnetcore.mvc.fromformattribute)** is inferred for action parameters of type [IFormFile](/dotnet/api/microsoft.aspnetcore.http.iformfile) and [IFormFileCollection](/dotnet/api/microsoft.aspnetcore.http.iformfilecollection).
```

Comment:
It would be good to add that `[FromForm]` won't be inferred for any simple types. If you want to bind form data in an api controller, you should tell us about it because it's not super common